src/audio/kpb.c

```diff
@@ -1593,6 +1594,44 @@ static enum task_state kpb_draining_task(void *arg)
 	return SOF_TASK_STATE_COMPLETED;
 }
 
+void kpb_convert_24b_to_32b(const void *linear_source, int ioffset,
+			    struct audio_stream __sparse_cache *sink, int ooffset,
+			    unsigned int n_samples)
+{
+	int ssize = audio_stream_sample_bytes(sink);
+	uint8_t *in = (uint8_t *)linear_source + ioffset * ssize;
+	uint8_t *out = audio_stream_wrap(sink, (uint8_t *)sink->w_ptr + ooffset * ssize);
+	ae_int32x2 *out_ptr = (ae_int32x2 *)(out);
```

Comment:
please add also a GCC version - without HiFi specific code

Reply:
ack - we separate the arch generic/hifi code using different files or with ifdefs.
@andrula-song @singalsu pls review the HiFi code.